frontend/src/main/scala/akkaviz/frontend/components/HierarchyPanel.scala

```diff
@@ -0,0 +1,100 @@
+package akkaviz.frontend.components
+
+import akkaviz.frontend.ActorRepository.ActorState
+import akkaviz.frontend.FrontendUtil
+import akkaviz.protocol
+import org.querki.jquery.JQueryStatic
+import org.scalajs.dom._
+import rx.Var
+
+import scala.scalajs.js.Dictionary
+import scalatags.JsDom.all._
+
+class HierarchyPanel(
+  currentActorState: (String) => Var[ActorState],
+  upstreamSend: protocol.ApiClientMessage => Unit
+)
+    extends Component {
+
+  private val $ = JQueryStatic
+
+  val seenActors: Dictionary[Unit] = Dictionary()
+
+  private val ActorAttr = "actor-path".attr
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [List(ActorAttr) should have a smaller scope by using private[this]](https://www.codacy.com/app/lustefaniak/akka-viz/file/2204035518/issues/source?bid=3121282&fileBranchId=3170129#l23)